src/main/java/com/google/edith/servlets/UserStatsServlet.java

```diff
@@ -0,0 +1,84 @@
+package com.google.edith.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns some example content. */
+@WebServlet("/user-stats-servlet")
+public class UserStatsServlet extends HttpServlet {
+
+  private final DatastoreService datastore;
+  private final UserInsightsInterface userInsights;
+
+  public UserStatsServlet() {
+    this.datastore = DatastoreServiceFactory.getDatastoreService();
+    this.userInsights = new UserInsightsService();
+  }
+
+  public UserStatsServlet(DatastoreService datastore, UserInsightsInterface userInsights) {
+    this.datastore = datastore;
+    this.userInsights = userInsights;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/json");
+    String userId = "userId";
+    response.getWriter().println(userInsights.createJson(userId));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    StringBuilder stringBuilder = new StringBuilder();
+    BufferedReader reader = request.getReader();
+    try {
+      String line;
+      while ((line = reader.readLine()) != null) {
+        stringBuilder.append(line).append("\n");
```

Comment:
not totally sure, but I think a try with resources could be used here instead
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
This change is from PR #23